### ISSUE TITLE
Vague documentation of Traits, Modifiers, and their interactions

### ISSUE THREAD

**Issue by Xuerian** [SlimeKnights/TinkersConstruct#2617]
It's great that bows are back, and I really like the new tool trait system.

However, trying to figure out how different properties interact if you don't already know is pretty rough.

(This was here before I noticed that in 2.6.0, luck only goes on bolts. Which is actually unfortunate because it means you can't luck the melee hits, but it is *definitely* more sensible)
> Say I want a crossbow that applies luck enemy drops?
> 
> Well, put that on the crossbow itself, according to a single google result on a forum.
> 
> How about extra damage for it?
> 
> Apparently that goes on the bolts, because while luck applies to shot enemies, sharpness only applies to crossbow melee.

There's no indication of how things like Manyullyn's traits apply, for example. And not even search results for how they work from other players who can understand how they're written.

What about the descriptions, too? Having a cute name for them is fun, but trying to figure out exactly what something does often isn't. Having a simple formula breakdown for traits and modifiers somewhere would be very helpful. Like, is someone supposed to know that pig iron adds a 1-in-200 bacon drop to broken blocks?

Documentation is really boring, which sucks. I get that 100%, but a lot of the traits or modifiers don't even have a comment in their source file explaining the behavior.

**Comment by bonii-xx** [comment]
> During combat you deal more and more damage but also consume more and more durability.

Seems clear enough, for exact details people will always have to experiment. The luck thing is more or less a makeshift solution. While the book states that the modifier has to go onto the damage dealing thing, people will still put it onto the bow and wonder why it doesn't work.

**Comment by Xuerian** [comment]
@bonii-xx If exploration is the point I can't really argue intent, even if I disagree in parts.

With some more documentation reading I understand the trait definitions better, so that more or less solves my own problem anyway. Works out.

**Comment by bonii-xx** [comment]
Which traits specifically are giving you a hard time? Maybe some are badly worded. Also by documentation I assume you mean the tooltips? Or external documentation?

**Comment by Xuerian** [comment]
I meant I spent more time reading the source and figuring out how effects/potions worked, which stumped me the first time. Documentation and definitions were not great word choices on my part, sorry.

As an ARPG and MMO player, some degree of minmaxing is always going to come out with me, but some of the effects have very similar descriptions and approximate material difficulties but oddly different values. The simplest examples to me are netherrack and silver. As opposed to the two AntiMonsterType modifiers, which have the same bonus.
Other bonuses scale (Superheat), while most don't, and there's no indication which
Some bonuses differ significantly and have equally neutral descriptions (Cobalt part trait speed, 10%, and Manyullyn part trait, 50% bonus on first hit)

I don't think I would have been bothered enough to make a ticket if  something like MmmMmmMmm was already in the game by default, it's a little hard to reliably test damage and effects without it. (I've added it to my mod list now, though)

The swappable part system does improve the process at least for traits, and it's reasonable to say that understanding the actual relative worth of each thing is a reasonable motivation to experiment with different options.
